jenkins/http-parser.jenkinsfile

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env groovy
+
+import groovy.json.JsonOutput
+
+pipeline {
+    agent any
+
+    parameters {
+        booleanParam(name: 'CERTIFY_SAFE', defaultValue: false, description: 'I have reviewed *the latest version of* these changes and I am sure that they don’t contain any code that could compromise the security of the CI infrastructure.')
+        string(defaultValue: "", description: 'The numeric ID of the pull request (from GitHub URL)', name: 'PR_ID')
+    }
+
+    stages {
+        stage('Clone repository') {
+            steps {
+                git 'https://github.com/nodejs/http-parser.git'
+            }
+        }
+
+        stage('Setup Git repository') {
+            when {
+                expression {
+                    return !!env.PR_ID
+                }
+            }
+
+            steps {
+                sh "git fetch origin pull/${env.PR_ID}/head:totest"
+                sh 'git checkout totest'
```

Comment:
Do these two need to be a bit more robust; like what if `totest` exists locally already and has a set of commits that doesn't match orign? What Does it need some `--force` or `--reset` or something?